Create ReadMe

kenpham205 opened the pull request.

kenpham205 committed:
Create ReadMe

kenpham205 committed:
Add files via upload

kenpham205 committed:
Delete SwerveCode directory

kenpham205 committed:
Add files via upload

kenpham205 committed:
Add files via upload

kenpham205 committed:
Add files via upload

kenpham205 committed:
Create ReadMe

kenpham205 committed:
Add files via upload

kenpham205 committed:
work  1/14/25

kenpham205 committed:
New code

MilkyTea123 commented on Swerve Motor Board Rev2.cydsn/CAN_Stuff.c:
```diff
@@ -28,12 +28,15 @@ void StartCAN(uint8 addy1, uint8 addy2) {
     } else address2 = addy2;
     
-    InitCAN(0x04, (int) address1, (int) address2);
+    InitCAN(0x04, (int) address1, (int) address2, (int) address3,(int) address4,(int) address5);
```
Device group shouthe ld be for science group. Should be one serial number? based on design

MilkyTea123 replied:
Device group should be the ld be for science group. Should be one serial number? based on design

MilkyTea123 commented on Swerve Motor Board Rev2.cydsn/CAN_Stuff.c:
```diff
@@ -60,8 +63,12 @@ int ProcessCAN(CANPacket* receivedPacket, CANPacket* packetToSend) {
     // if ()
     // if ()
     
+    MOTORS_TO_CHANGE = motor_adress
```
MISSPELLED :(

jkljason replied:
Fixed

MilkyTea123 commented on Swerve Motor Board Rev2.cydsn/CAN_Stuff.c:
```diff
@@ -28,12 +28,15 @@ void StartCAN(uint8 addy1, uint8 addy2) {
+    if (motor == MOTOR4) return address4;
+    if (motor == MOTOR5) return address5;
     return 0;
 }
```
make sure that you're able to have multiple motors changing at once

MilkyTea123 replied:
make sure that you're able to have multiple motors changing at once

MilkyTea123 commented on Swerve Motor Board Rev2.cydsn/CAN_Stuff.c:
```diff
@@ -60,8 +63,12 @@ int ProcessCAN(CANPacket* receivedPacket, CANPacket* packetToSend) {
     // if ()
     // if ()
```
can remove commented out stuff unless its documentation

MilkyTea123 commented on Swerve Motor Board Rev2.cydsn/CAN_Stuff.c:
```diff
@@ -60,8 +63,12 @@ int ProcessCAN(CANPacket* receivedPacket, CANPacket* packetToSend) {
+    else if (motor_address == address3) motor = MOTOR3;
+    else if (motor_address == address4) motor = MOTOR4;
+    else if (motor_address == address5) motor = MOTOR5;
     else motor = MOTOR_BOTH; // assume broadcast
```
make sure to match these changes with the ones above if you make them

MilkyTea123 commented on Swerve Motor Board Rev2.cydsn/FSM_Stuff.c:
```diff
@@ -15,7 +15,7 @@
 #include "PositionPID.h"
 
-int mode1, mode2;
+int mode1, mode2, mode3, mode4, mode5;
```
can change to array

MilkyTea123 commented on Swerve Motor Board Rev2.cydsn/FSM_Stuff.c:
```diff
@@ -35,11 +35,14 @@ int SetMode(int motor, int new_mode) {
+    //    if (motor(i) == 1) 
     /* case (motor(1))
             motor(1) == 0: turn off
             motro(1) == 1: turn on
```
can remove commented code unless documentation

MilkyTea123 commented on Swerve Motor Board Rev2.cydsn/MotorDrive.c:
```diff
@@ -33,14 +33,23 @@ int32 PWM5_value = 0;
 volatile uint8  PWM2_invalidate = 0;
+volatile uint8  PWM3_invalidate = 0;
+volatile uint8  PWM4_invalidate = 0;
+volatile uint8  PWM5_invalidate = 0;
```
can change to array

jkljason committed:
Pin Assignments 3/4

kenpham205 committed:
y

kenpham205 committed:
fixed pins

jkljason committed:
Jason Update through break for 5 motors

jkljason committed:
Added PCA9685 Header and C from servo board

jkljason committed:
PCA9685 Files from Servo

jkljason committed:
Renamed library and used driver. 3 firmware errors left

kenpham205 committed:
fixed code

jkljason committed:
Can finally build

Commented out pin limit. Also replaced read dip with hardcoded 1, 17 for dip values on startCAN function. Included functions with new PCA9685 driver

jkljason committed:
Merge branch 'kenpham205-patch-2' of https://github.com/huskyroboticsteam/Science_DC_Board into kenpham205-patch-2
